agent/src/util/discovery_operator.rs

```diff
@@ -206,13 +214,25 @@ impl DiscoveryOperator {
     ) -> anyhow::Result<()> {
         // clone objects for thread
         let discovery_operator = Arc::new(self.clone());
-        let stop_discovery_receiver: &mut tokio::sync::broadcast::Receiver<()> =
+        let stop_discovery_receiver: &mut tokio::sync::broadcast::Receiver<Option<ConfigId>> =
             &mut dh_details.close_discovery_handler_connection.subscribe();
         loop {
             // Wait for either new discovery results or a message to stop discovery
             tokio::select! {
-                _ = stop_discovery_receiver.recv() => {
-                    trace!("internal_do_discover - received message to stop discovery for endpoint {:?} serving protocol {}", dh_details.endpoint, discovery_operator.get_config().spec.discovery_handler.name);
+                result = stop_discovery_receiver.recv() => {
+                    if let Ok(Some(config_id)) = result {
```

Comment:
add comment saying something like `stop is triggered iff None is sent (in which case, all are stopped) or if this config_id is sent (in which case, only this is stopped)`